kotlinx-coroutines-core/native/src/Builders.kt

```diff
@@ -56,21 +58,23 @@ private class BlockingCoroutine<T>(
         get() = false // it throws exception to parent instead of cancelling it
 
     @Suppress("UNCHECKED_CAST")
-    fun joinBlocking(): T {
+    fun joinBlocking(): T = memScoped {
         try {
             eventLoop?.incrementUseCount()
+            val timespec = alloc<timespec>()
             while (true) {
                 val parkNanos = eventLoop?.processNextEvent() ?: Long.MAX_VALUE
                 // note: process next even may loose unpark flag, so check if completed before parking
                 if (isCompleted) break
-                // todo: LockSupport.parkNanos(this, parkNanos)
+                timespec.tv_nsec = parkNanos.convert()
```

Comment:
tv_nsec should not be greater then 999999999.
Therefore you should set tv_sec as well right ?
```kotlin
timespec.tv_sec = parkNanos / 1000000000L
timespec.tv_nsec = (parkNanos % 1000000000L).convert()
```

Reply:
good catch, thanks!